### Importing libraries

In [54]:
import numpy as np
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import cross_val_score
from sklearn.metrics import make_scorer, mean_squared_error, mean_absolute_error
from sklearn.model_selection import KFold, train_test_split
import pandas as pd
import matplotlib.pyplot as plt
import matplotlib
matplotlib.style.use('ggplot')
from Utils import *

%matplotlib inline

### Loading the training data

In [2]:
train = np.load('../Models/train.npy') # we only want to work with the training data
X = train[0:,0:-1]
y = train[:, -1]

In [3]:
print "X: ", X.shape
print "y: ", y.shape

X:  (3184, 49)
y:  (3184,)


### Train-Test Split

In [4]:
X_train = np.load("../Models/X_train.npy")
X_test = np.load("../Models/X_test.npy")
y_train = np.load("../Models/y_train.npy")
y_test = np.load("../Models/y_test.npy")

In [5]:
print X_train.shape
print X_test.shape
print y_train.shape
print y_test.shape

(2547, 49)
(637, 49)
(2547,)
(637,)


## Build Baseline Model 

In [55]:
baseline_linreg = LinearRegression()

##  Stratified 10-Fold Cross Validation with Linear Regression

In [56]:
from sklearn.model_selection import cross_val_score

In [60]:
print "Average MAE: ", np.mean(cross_val_score(estimator=LinearRegression(), X=X, y=y, scoring=mae_scorer_cv, cv=10))
print "Average RMSE: ", np.mean(cross_val_score(estimator=LinearRegression(), X=X, y=y, scoring=rmse_scorer_cv, cv=10))

Average MAE:  3.61816601526
Average RMSE:  4.66181876342


##  Baseline Model Scores
 * Average CV MAE: 3.618
 * Average CV RMSE: 4.6618
 